client/state/products-list/actions.js

```diff
@@ -37,23 +37,35 @@ export function receiveProductsList( productsList, type = null ) {
  * 								or undefined, for all products
  * @returns {Function} 			an Action thunk
  */
-export function requestProductsList( query = {} ) {
+export function requestProductsList( query = {}, siteQueryFailed = false ) {
+	const queryParams = new URLSearchParams( window.location.search );
+	const site = siteQueryFailed ? null : queryParams.get( 'site' );
```

Comment:
@CodeyGuyDylan Reading the `site` param from `window.location` is not a very good idea: it introduces hidden dependencies that are hard to track. On which routes is this `site` param used? It's impossible to tell.

Can we make `site` an explicit param of `requestProductList` and add a `site` prop to the `QueryProductsList` component that issues the request. That would be much better.

Reply:
Our team is working on an urgent project at the moment, but I will add this to our maintenance board to look at. [Here is the issue](https://github.com/Automattic/jetpack-marketing/issues/1226)